flow/project.py

```diff
@@ -869,9 +917,12 @@ def isdisjoint(self, group):
         """
         return set(self).isdisjoint(set(group))
 
-    def _generate_id(self, job, operation_name=None, index=0):
+    def _generate_id(self, jobs, operation_name=None, index=0):
         "Return an id, which identifies this group with respect to this job."
-        project = job._project
+        project = jobs[0]._project
+
+        for job in jobs:
+            assert job._project == project
```

Comment:
This might be worth being a static method of the class. Something like `_verify_aggregate`.

Reply:
We only do it here and check in other places that require it through `_generate_id`.